pkg/chain/ethlike/chain.go

```diff
@@ -0,0 +1,93 @@
+package ethlike
+
+import (
+	"context"
+	"encoding/hex"
+	"math/big"
+)
+
+// Hash represents the 32 byte Keccak256 hash of arbitrary data.
+type Hash [32]byte
+
+func (h Hash) String() string {
+	return hex.EncodeToString(h[:])
+}
```

Comment:
It's used implicitly while logging tx hash. Specifically in the mining waiter. I'm looking at this as a generic chain-agnostic way of displaying an `ETH-like` address.  Naming it `TerminalString` will tie it to `go-ethereum`.

Reply:
What I like about `TerminalString()` is that it makes it clear that the value returned by the function should not be used for testing equality or used as a parameter to some other function for a purpose other than logging.

That's the risk with just `String()` - we may one day be implementing adapter for another eth-like chain and find that the delegate chain function for `TransactionReceipt` accepts hash as a string, we'll do `String()` because why not and we'll end up with a bug or not working code.

IMO it is important to make it very clear that to-string transformations of anything in `ethlike` are not chain-agnostic, especially in the context of ECDSA client chain interface refactoring where all identifiers implement `fmt.Stringer`.

Reply:
Ok that's make sense. Done in https://github.com/keep-network/keep-common/commit/01bb2719c08f4616f00486ccb7fd0f8807e0a289. Also, changed the internal implementation to do the same as `go-ethereum` does and produce shorter strings.